packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -986,15 +986,21 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
         minimumSize: MaterialStateProperty.resolveWith<Size?>((Set<MaterialState> states) {
           final double? effectiveMaximumWidth =
               effectiveMenuStyle!.maximumSize?.resolve(states)?.width;
-          return Size(math.min(widget.width!, effectiveMaximumWidth ?? 0.0), 0.0);
+          final double effectiveWidth = effectiveMaximumWidth == null
+            ? widget.width!
+            : math.min(widget.width!, effectiveMaximumWidth);
+          return Size(effectiveWidth, 0.0);
         }),
       );
     } else if (anchorWidth != null) {
       effectiveMenuStyle = effectiveMenuStyle.copyWith(
         minimumSize: MaterialStateProperty.resolveWith<Size?>((Set<MaterialState> states) {
           final double? effectiveMaximumWidth =
               effectiveMenuStyle!.maximumSize?.resolve(states)?.width;
-          return Size(math.min(anchorWidth, effectiveMaximumWidth ?? 0.0), 0.0);
+          final double effectiveWidth = effectiveMaximumWidth == null
+            ? anchorWidth
+            : math.min(anchorWidth, effectiveMaximumWidth);
+          return Size(effectiveWidth, 0.0);
```

Comment:
And this is the same as `Size(math.min(anchorWidth, effectiveMaximumWidth ?? anchorWidth), 0.0);`